uiux/challenge-1/src/models/artist.tsx

```diff
@@ -0,0 +1,43 @@
+import Track from './track';
+import { searchTracksByArtistId } from '../services/musicmatch';
+
+export default class Artist {
+    name: string;
+    id: number;
+    tracksFetched: boolean;
+    tracks: Track[];
+
+    constructor(data) {
+        let artist = data;
+        if (data.artist) {
+            artist = data.artist;
+        }
+        this.name = artist.artist_name;
+        this.id = artist.artist_id;
+        this.tracks = [];
+        this.tracksFetched = false;
+    }
+    
+    getTracks() {
+        if (!this.tracksFetched) {
```

Comment:
An artist could not have any tracks, so to avoid making a query again, I simply set a boolean to avoid unnecessary request to the API